sky/serve/replica_managers.py

```diff
@@ -616,13 +618,17 @@ def _download_and_stream_logs(info: ReplicaInfo):
             assert isinstance(handle, backends.CloudVmRayResourceHandle)
             replica_job_logs_dir = os.path.join(constants.SKY_LOGS_DIRECTORY,
                                                 'replica_jobs')
-            log_file = controller_utils.download_and_stream_latest_job_log(
-                backend, handle, replica_job_logs_dir)
-            if log_file is not None:
+            job_log_file_name = (
+                controller_utils.download_and_stream_latest_job_log(
+                    backend, handle, replica_job_logs_dir))
+            if job_log_file_name is not None:
                 logger.info(f'\n== End of logs (Replica: {replica_id}) ==')
-                with open(local_log_file_name,
-                          'a') as local_file, open(log_file, 'r') as job_file:
-                    local_file.write(job_file.read())
+                with open(log_file_name, 'a') as replica_log_file, open(
+                        job_log_file_name, 'r') as job_file:
+                    replica_log_file.write(job_file.read())
+            else:
+                logger.error(f'Failed to sync down logs from replica'
```

Comment:
actually i mean to write some stuff to the file; so later when the user tail the log they could see the error message again. we already have logging info in the stream_and_download function

Reply:
Ah right, I misunderstood before. Fixed!